stackslib/src/chainstate/burn/operations/stack_stx.rs

```diff
@@ -322,16 +357,26 @@ impl StacksMessageCodec for PreStxOp {
 
 impl StacksMessageCodec for StackStxOp {
     /*
-            Wire format:
-            0      2  3                             19        20
-            |------|--|-----------------------------|---------|
-             magic  op         uSTX to lock (u128)     cycles (u8)
+            0      2  3                             19           20    21                 54
+            |------|--|-----------------------------|------------|-----|-------------------|
+            magic  op         uSTX to lock (u128)     cycles (u8) option     signer key
+                                                                  marker
     */
     fn consensus_serialize<W: Write>(&self, fd: &mut W) -> Result<(), codec_error> {
         write_next(fd, &(Opcodes::StackStx as u8))?;
         fd.write_all(&self.stacked_ustx.to_be_bytes())
             .map_err(|e| codec_error::WriteError(e))?;
         write_next(fd, &self.num_cycles)?;
+
+        if let Some(signer_key) = self.signer_key {
+            fd.write_all(&(1 as u8).to_be_bytes())
+                .map_err(|e| codec_error::WriteError(e))?;
+            fd.write_all(&signer_key.as_bytes()[..])
+                .map_err(codec_error::WriteError)?;
+        } else {
+            fd.write_all(&(0 as u8).to_be_bytes())
+                .map_err(|e| codec_error::WriteError(e))?;
+        }
```

Comment:
Relating to my other comment, we need to make sure this is backwards compatible. Otherwise, the same exact op would serialize differently. Because of that, I wonder if it's better to just drop the option flag. If it weren't for backwards compatibility I think the option flag is good, but it might add too much complexity in this case.

Reply:
TBH I didn't consider the consensus-breaking change and just mirrored what was done for delegate-stx for two of its optional fields...I fixed this now by going with your second suggestion since it didn't require an epoch check.